File: TelerikUIExamples/TelerikUIExamples/LineAreaSpline.m
//
//  LineAreaSpline.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "LineAreaSpline.h"
#import <TelerikUI/TelerikUI.h>

@implementation LineAreaSpline
{
    TKChart *_chart;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self addOption:@"Line" selector:@selector(reloadChart)];
        [self addOption:@"Spline" selector:@selector(reloadChart)];
        [self addOption:@"Area" selector:@selector(reloadChart)];
        [self addOption:@"Area Spline" selector:@selector(reloadChart)];
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:_chart];
    
    [self reloadChart];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)reloadChart
{
    [_chart removeAllData];
    
    for (int i = 0; i<3; i++) {
        NSMutableArray *array = [[NSMutableArray alloc] init];
        for (int i = 0; i<7; i++) {
            [array addObject:[[TKChartDataPoint alloc] initWithX:@(i+1) Y:@(arc4random() % 100)]];
        }
        
        TKChartSeries *series = nil;
        switch (self.selectedOption) {
            case 0:
                series = [[TKChartLineSeries alloc] initWithItems:array];
                break;
                
            case 1:
                series = [[TKChartSplineSeries alloc] initWithItems:array];
                break;
                
            case 2:
                series = [[TKChartAreaSeries alloc] initWithItems:array];
                break;
                
            case 3:
                series = [[TKChartSplineAreaSeries alloc] initWithItems:array];
                break;
        }
        series.selectionMode = TKChartSeriesSelectionModeSeries;
        [_chart addSeries:series];
    }

    [_chart reloadData];
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/LineAreaSpline.swift
//
//  LineAreaSpline.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class LineAreaSpline: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Line", action: reloadChart)
        self.addOption("Spline", action: reloadChart)
        self.addOption("Area", action: reloadChart)
        self.addOption("Area Spline", action: reloadChart)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        self.reloadChart()
    }
    
    func reloadChart() {
        chart.removeAllData()
        
        for _ in 0..<3 {
            var items = [TKChartDataPoint]()
            for i in 0..<8 {
                items.append(TKChartDataPoint(x:(i+1), y:Int(arc4random()%100)))
            }
            var series:TKChartSeries
            switch self.selectedOption {
                case 1:
                    series = TKChartSplineSeries(items:items)
                    break
            
                case 2:
                    series = TKChartAreaSeries(items:items)
                    break
            
                case 3:
                    series = TKChartSplineAreaSeries(items:items)
                    break
                
                default:
                    series = TKChartLineSeries(items:items)
                    break;
            }
            series.selectionMode = TKChartSeriesSelectionMode.Series
            chart.addSeries(series)
        }
        chart.reloadData()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
    
